### ISSUE TITLE
Improve npm support

### ISSUE THREAD

**Issue by ai** [LeaVerou/color.js#158]
What we can do to improve DX for developers using tools from npm.

To be honest, right now there is a feeling that it is CDN-first tool.

Easy to implement:

1. Add ChangeLog. I use `ChangeLog.md`. Other projects use GitHub releases. Document any small release.
2. After accepting a PR explain when it will be released. I mostly release PR imminently or explain that I am waiting some other PRs before the release.
3. Seems like we need SemVer. On `0.x` branch, I use `0.X.x` releases for new features (or breaking changes) and `0.x.X` releases for fixes. For instance, 0.0.7 release added OKLCH and should be 0.1.
4. Add small docs to `README.md` (at least main example and short description) to have better [npm page](https://www.npmjs.com/package/colorjs.io).

Harder to implement:

1. Rename tool to have memorable name on npm. It was nice to have the same name in npm and in the repo.

Questionable:

1. Move website to another repo or at least subfolder?

**Comment by LeaVerou** [comment]
Hey @ai, thanks for posting this! AFAIK both @svgeesus and I primarily use it from the CDN but we do care about meeting the needs of Node.js devs since there are so many use cases for Color.js that involve Node. So any help is welcome!

1. We already do: https://github.com/LeaVerou/color.js/releases Is there something we should do differently?
2. Sure, that's easy to do!
3. I also learned this recently, but according to [the semver spec](https://semver.org/#spec-item-4):

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

That said, we did increase the minor version (`0.0.7` --> `0.1.0` recently when we made a release with a ton of breaking changes, and will try to follow that pattern more closely post-release.

4. Oh yeah that's a good point. We could just port some of the docs in the homepage to README.md, then use that when building the homepage to eliminate duplication.

Changing the name is probably a no-go at this point. But maybe there's a better (available) package name than `colorjs.io`?

You mean move the repo to an org? That's definitely the plan down the line.

**Comment by LeaVerou** [comment]
Oh I see what you mean with the last point, move the *website* to another repo or subfolder.

The problem with that is that I want the website to always run on the latest commit (even if unreleased) which becomes rather complicated when the website is in a different repo or even subfolder. If you have any suggestions for how to handle that, I'd *love* to have the website in a different repo or a subfolder!

**Comment by ai** [comment]
> We already do: https://github.com/LeaVerou/color.js/releases Is there something we should do differently?

Yes, it is exactly what I looked for. Maybe I missed it for some reason.

> The problem with moving the website elsewhere is that I want it to always run on the latest commit of Color.js (even if unreleased) which becomes rather complicated when the website is in a different repo or even subfolder.

What is a problem with subfolder?

**Comment by LeaVerou** [comment]
I'm glad we are discussing this because this is a problem I've had with all of my libraries, and none of the solutions I've gone with over the years seems good to me.

> What is a problem with subfolder?

If `dist/` stays out of the subfolder, then the website cannot link to it. I suppose I could move `dist` into the subfolder. Is that common though? I think `dist/` is almost always in the root, no? 
Also there are parts of the website that import individual modules instead of the whole thing (e.g. tests).
And right now the website is also the CDN: people can do things like:

```js
import Color from "https://colorjs.io/src/color.js";
import "https://colorjs.io/src/spaces/p3.js";
import "https://colorjs.io/src/spaces/rec2020.js";
import "https://colorjs.io/src/deltaE/deltaE2000.js";
```

So, I'd still need to deploy something that includes `src` somewhere, but I suppose I could just set up another Netlify website for that (e.g. `src.colorjs.io` or `get.colorjs.io`).

Stepping back for a moment, is the primary problem of having the website files mixed with the library files the messiness, or that it's included in `node_modules`? Cause if the latter, we can just remove everything except `src` and `dist` via .npmignore.

**Comment by ai** [comment]
> If dist/ stays out of the subfolder, then the website cannot link to it. I suppose I could move dist into the subfolder. Is that common though? I think dist/ is almost always in the root, no?

Or you may use some plugin for 11ty to copy `dist` to `webite/dist`.

**Comment by LeaVerou** [comment]
@ai should we close it or rename it to "Move website to separate subfolder or repo"? That's the only "open" issue out of the list in the original post.

**Comment by LeaVerou** [comment]
(closing was accidental due to the GitHub keyboard shortcut regression, I keep accidentally closing issues today, hope they fix it soon 😢 )

**Comment by ai** [comment]
Done. Renamed.
